### PR TITLE
#347 - references order, projects order, references position

### LINKED ISSUE #347: References order
I updated one of references and it changed order in dashboard's index table and on home page. We should be able to manage how they are displayed on home page at least.

We could also add person name to dashboard index table.

### PR DESCRIPTION
**Pull request by AleksandraKozubal**
In this pr i added possibility to change references order along with projects order and comment suggestions - added position in references.

In admin panel user can click on icon to change order:
image

image

Position displaying:
image

It should close #347.

### CODE REVIEW

File: resources/views/components/reference.blade.php [hunk 1]
```diff
@@ -8,7 +8,12 @@
              alt="{{ __('alt.reference') . ' ' . $reference->creator_name }}">
         <div>
             <div class="font-semibold text-left">{{ $reference->creator_name }}</div>
-            <div class="text-gray-600 text-left">{{ $reference->company }}</div>
+            <div class="text-gray-600 text-left">
+                @isset( $reference->position ){{ $reference->position }}@endisset
+                @if( isset($reference->position) && isset($reference->company) ){{ __('at') }}@endif
+                @isset( $reference->company ){{ $reference->company }}@endisset
```
Comment (krzysztofrewak) [inline resources/views/components/reference.blade.php:14]:
```suggestion
                @isset($reference->position) {{ $reference->position }} @endisset
                @if(isset($reference->position) && isset($reference->company)) {{ __('at') }} @endif
                @isset($reference->company) {{ $reference->company }} @endisset
```

File: resources/views/components/references.blade.php [hunk 2]
```diff
@@ -104,38 +104,52 @@
...
                          src="{{ $references[$main]->getPhotoPath() }}"
                          alt="{{ __('alt.reference') . " " . $references[$main]->creator_name }}">
                     <div class="flex-auto">
                         <div class="font-semibold text-left">{{ $references[$main]->creator_name }}</div>
-                        <div class="text-gray-600 text-left">{{ $references[$main]->company }}</div>
+                        <div
+                            class="text-gray-600 text-left">
+                            @isset( $references[$main]->position ){{ $references[$main]->position }}@endisset
+                            @if( isset($references[$main]->position) && isset($references[$main]->company) ){{ __('at') }}@endif
+                            @isset( $references[$main]->company ){{ $references[$main]->company }}@endisset
```
Comment (krzysztofrewak) [inline resources/views/components/references.blade.php:129]:
```suggestion
                            @isset($references[$main]->position) {{ $references[$main]->position }} @endisset
                            @if(isset($references[$main]->position) && isset($references[$main]->company)) {{ __('at') } @endif
                            @isset($references[$main]->company) {{ $references[$main]->company }} @endisset
```

File: resources/views/components/references.blade.php [hunk 3]
```diff
@@ -104,38 +104,52 @@
...
                          src="{{ $references[$main2]->getPhotoPath() }}"
                          alt="{{ __('alt.reference') . " " . $references[$main2]->creator_name }}">
                     <div class="flex-auto">
                         <div class="font-semibold text-left">{{ $references[$main2]->creator_name }}</div>
-                        <div class="text-gray-600 text-left">{{ $references[$main2]->company }}</div>
+                        <div class="text-gray-600 text-left">
+                            @isset( $references[$main2]->position ){{ $references[$main2]->position }}@endisset
+                            @if( isset($references[$main2]->position) && isset($references[$main2]->company) ){{ __('at') }}@endif
+                            @isset( $references[$main2]->company ){{ $references[$main2]->company }}@endisset
+                        </div>
```
Comment (krzysztofrewak) [inline resources/views/components/references.blade.php:152]:
```suggestion
                            @isset($references[$main2]->position) {{ $references[$main2]->position }} @endisset
                            @if(isset($references[$main2]->position) && isset($references[$main2]->company)) {{ __('at') }}@endif
                            @isset($references[$main2]->company) {{ $references[$main2]->company }} @endisset
                        </div>
```

Comment (krzysztofrewak) [reply resources/views/components/references.blade.php:152]: These block are repeated. Can you make a component from that?

Comment (krzysztofrewak) [reply resources/views/components/references.blade.php:152]: Also: are we safe when there'd be no `$main2` index in `$references`?

File: app/Models/Reference.php [hunk 4]
```diff
@@ -15,6 +15,8 @@
  * @property string $company
```
Comment (krzysztofrewak) [inline app/Models/Reference.php:15]:
```suggestion
 * @property ?string $company
```

File: app/Models/Project.php [hunk 5]
```diff
@@ -17,6 +17,7 @@
  * @property string $template
  * @property bool $published
  * @property string $color
+ * @property int $sort_order
```
Comment (krzysztofrewak) [inline app/Models/Project.php:20]:
```suggestion
 * @property ?int $sort_order
```

Comment (krzysztofrewak) [reply app/Models/Project.php:20]: Or maybe it should have been defaulted by some number (like zero?) in database migration?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Can we change icons to some arrows? And this text "Zmień kolejność przeciągając." to something like "Przytrzymaj i przeciągnij strzałki, aby zmienić kolejność."? 

2. When I change project order I noticed that in column "Nazwa realizacji" we don't have whole title, just first part. Can we make that the whole title is shown?

3. When position and company is too long, the reference looks ugly

image

File: resources/views/components/references.blade.php (hunk 3 above)
Comment (AleksandraKozubal) [reply resources/views/components/references.blade.php]: > Also: are we safe when there'd be no `$main2` index in `$references`?

Yes, we are handling that.

File: app/Models/Project.php (hunk 5 above)
Comment (AleksandraKozubal) [reply app/Models/Project.php]: Added nullable symbol, I think that it's ok to have it nullable, there is no problem with applying order.

**Comment by AleksandraKozubal** [comment]
> 1. Can we change icons to some arrows? And this text "Zmień kolejność przeciągając." to something like "Przytrzymaj i przeciągnij strzałki, aby zmienić kolejność."?
> 
> 2. When I change project order I noticed that in column "Nazwa realizacji" we don't have whole title, just first part. Can we make that the whole title is shown?
> 
> 3. When position and company is too long, the reference looks ugly
> 
> 
> [image](https://private-user-images.githubusercontent.com/56546832/366323939-b38c5254-5957-4341-bbe6-17b9405d4b75.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwNTQ2NDAsIm5iZiI6MTcyNjA1NDM0MCwicGF0aCI6Ii81NjU0NjgzMi8zNjYzMjM5MzktYjM4YzUyNTQtNTk1Ny00MzQxLWJiZTYtMTdiOTQwNWQ0Yjc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTExVDExMzIyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4ODczN2I2YzZkYjJhODMzMTAyZWUwZjNmMzkxNDU3NDJlMDRmNWE2NDc1OWY5ZmRhN2VmMDhkYjE4NjVhOGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.P8G7JOn-bS0UsIFzcBca6t-055y9rWCGwN7AVEo53xA)

1. Icons and text are provided by filament, it would need a lot of work for such a little thing, so I would stick to that
2. In one cell can be only one record shown - our name is splitted into three records, so it would look like that:
image
3. That's because you provided one long string with no spaces, when data is provided as normal it looks allright

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Review by krzysztofrewak — APPROVED** [review]

**Pull request merged**